src/telemetry/Manager.cc

```diff
@@ -38,6 +43,232 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
+
+	return r;
+	}
+
+namespace bt = broker::telemetry;
+
+/**
+ * A collector implementation fetching info and values (for counters and guages);
+ */
+class MetricsCollector : public broker::telemetry::metrics_collector
+	{
+	using MT = Manager::Metric::MetricType;
+
+public:
+	MetricsCollector(std::string_view prefix, std::string_view name)
+		: prefix_pattern_(prefix), name_pattern_(name)
+		{
+		}
+
+	void operator()(const bt::dbl_counter_family_hdl* family, const bt::dbl_counter_hdl* counter,
+	                const std::vector<std::string_view>& label_values)
+		{
+		if ( matches(upcast(family)) )
+			metrics_.push_back({MT::DOUBLE_COUNTER, bt::upcast(family), (double)bt::value(counter),
+			                    std::nullopt, label_values});
+		}
+
+	void operator()(const bt::int_counter_family_hdl* family, const bt::int_counter_hdl* counter,
+	                const std::vector<std::string_view>& label_values)
+		{
+		if ( matches(upcast(family)) )
+			metrics_.push_back({MT::INT_COUNTER, bt::upcast(family), (double)bt::value(counter),
+			                    bt::value(counter), label_values});
+		}
+
+	void operator()(const bt::dbl_gauge_family_hdl* family, const bt::dbl_gauge_hdl* gauge,
+	                const std::vector<std::string_view>& label_values)
+		{
+		if ( matches(upcast(family)) )
+			metrics_.push_back({MT::DOUBLE_GAUGE, bt::upcast(family), (double)bt::value(gauge),
+			                    std::nullopt, label_values});
+		}
+
+	void operator()(const bt::int_gauge_family_hdl* family, const bt::int_gauge_hdl* gauge,
+	                const std::vector<std::string_view>& label_values)
+		{
+		if ( matches(upcast(family)) )
+			metrics_.push_back({MT::INT_GAUGE, bt::upcast(family), (double)bt::value(gauge),
+			                    bt::value(gauge), label_values});
+		}
+
+	void operator()(const bt::dbl_histogram_family_hdl* family,
+	                const bt::dbl_histogram_hdl* histogram,
+	                const std::vector<std::string_view>& label_values)
+		{
+		if ( matches(upcast(family)) )
+			metrics_.push_back({MT::DOUBLE_HISTOGRAM, bt::upcast(family), std::nullopt,
+			                    std::nullopt, label_values});
+		}
+
+	void operator()(const bt::int_histogram_family_hdl* family,
+	                const bt::int_histogram_hdl* histogram,
+	                const std::vector<std::string_view>& label_values)
+		{
+		if ( matches(upcast(family)) )
+			metrics_.push_back(
+				{MT::INT_HISTOGRAM, bt::upcast(family), std::nullopt, std::nullopt, label_values});
+		}
+
+	const std::vector<Manager::Metric>& result() { return metrics_; }
+
+private:
+	/**
+	 * Does the given family match the prefix, name patterns?
+	 */
+	bool matches(const bt::metric_family_hdl* family)
+		{
+		// Complications due to bt::prefix and bt::name returning
+		// a string_view which isn't guaranteed to be null-terminated.
+		bool have_prefix = ! prefix_pattern_.empty();
+		bool have_name = ! name_pattern_.empty();
+
+		if ( have_prefix )
```

Comment:
What supposed to the result of prefix/name are empty? Should the matching fail then (i.e. do we need `else` branches?)

Reply:
The intention was to have an empty string match everything. If that feels wrong can certainly change and use `"*"` defaults in script-land instead.

Reply:
(would also remove  the awkward `empty()` test).